server/service/orbit.go

```diff
@@ -130,6 +132,10 @@ func (svc *Service) EnrollOrbit(ctx context.Context, hostInfo fleet.OrbitHostInf
 
 	secret, err := svc.ds.VerifyEnrollSecret(ctx, enrollSecret)
```

Comment:
@sharon-fdm The changes introduced here should fix the issue in question, but I think `svc.ds.VerifyEnrollSecret` should be refactored - the current implementation will return an error if the `SELECT team_id FROM enroll_secrets WHERE secret = ?` query returns no results, but it has been established that is not an error condition.  I think `svc.ds.VerifyEnrollSecret` should return `nil, nil` in that scenario, and then it would be the responsibility of the caller to figure out what to do if `secret` is nil. 

This was not done in this PR because it was outside the scope of the issue.

Reply:
Thanks for catching this @juan-fdz-hawa 
Can you open a ticket for this please?

Reply:
IMO current behavior of `Datastore.VerifyEnrollSecret` where it fails if the secret doesn't exist is less error prone.
(As long as it's well documented what error is returned when the secret doesn't exist).